earn/src/data/BorrowerNft.ts

```diff
@@ -0,0 +1,122 @@
+import { ContractCallContext, Multicall } from 'ethereum-multicall';
+import { BigNumber, ethers } from 'ethers';
+import { borrowerLensAbi } from 'shared/lib/abis/BorrowerLens';
+import { borrowerNftAbi } from 'shared/lib/abis/BorrowerNft';
+import {
+  ALOE_II_BORROWER_LENS_ADDRESS,
+  ALOE_II_BORROWER_NFT_ADDRESS,
+  MULTICALL_ADDRESS,
+} from 'shared/lib/data/constants/ChainSpecific';
+import { Address } from 'wagmi';
+
+type BorrowerNftFilterParams = {
+  validManagerSet?: Set<Address>;
+  validUniswapPool?: Address;
+  onlyCheckMostRecentModify: boolean;
+  includeFreshBorrowers: boolean;
+};
+
+export async function fetchListOfBorrowerNfts(
+  chainId: number,
+  provider: ethers.providers.BaseProvider,
+  userAddress: string,
+  filterParams?: BorrowerNftFilterParams
+) {
```

Comment:
Could we make the return type explicit?

Reply:
I don't really love this return type, though its functional for now. I don't really want to make it explicit because then we're less likely to come up with a better one in the future.